src/main/cpp/sim/run_stride.cpp

```diff
@@ -53,135 +58,205 @@ using namespace boost::property_tree;
+	if (ConfigInfo::HaveOpenMP()) {
+		cout << "Using OpenMP threads:  " << num_threads << endl;
+	} else {
+		cout << "Not using OpenMP threads." << endl;
+	}
+	return num_threads;
+}
+
+/// Prints information about the current execution environment.
+void print_execution_environment()
+{
+	cout << "\n*************************************************************" << endl;
+	cout << "Starting up at:      " << TimeStamp().ToString() << endl;
+	cout << "Executing:           " << InstallDirs::GetExecPath().string() << endl;
+	cout << "Current directory:   " << InstallDirs::GetCurrentDir().string() << endl;
+	cout << "Install directory:   " << InstallDirs::GetRootDir().string() << endl;
+	cout << "Data    directory:   " << InstallDirs::GetDataDir().string() << endl;
+}
+
+/// Verifies that Stride is being run in the right execution environment.
+void verify_execution_environment()
+{
+	if (InstallDirs::GetCurrentDir().compare(InstallDirs::GetRootDir()) != 0) {
+		throw runtime_error(string(__func__) + "> Current directory is not install root! Aborting.");
+	}
+	if (InstallDirs::GetDataDir().empty()) {
+		throw runtime_error(string(__func__) + "> Data directory not present! Aborting.");
+	}
+}
+
+/// Run the stride simulator.
+void run_stride(const MultiSimulationConfig& config)
+{
+	// -----------------------------------------------------------------------------------------
+	// OpenMP.
+	// -----------------------------------------------------------------------------------------
+	unsigned int num_threads = print_number_of_omp_threads();
+
+	// -----------------------------------------------------------------------------------------
+	// Set output path prefix.
+	// -----------------------------------------------------------------------------------------
+	auto output_prefix = config.log_config->output_prefix;
+	if (output_prefix.length() == 0) {
+		output_prefix = TimeStamp().ToTag();
+	}
+	cout << "Project output tag:  " << output_prefix << endl << endl;
+
+	// -----------------------------------------------------------------------------------------
+	// Track index case setting.
+	// -----------------------------------------------------------------------------------------
+	cout << "Setting for track_index_case:  " << boolalpha << config.common_config->track_index_case << endl;
+
+	// Set the log queue size.
+	spdlog::set_async_mode(1048576);
+
+	// -----------------------------------------------------------------------------------------
+	// Create simulator.
+	// -----------------------------------------------------------------------------------------
+	Stopwatch<> total_clock("total_clock", true);
+	multiregion::ParallelSimulationManager<StrideSimulatorResult, size_t> sim_manager{num_threads};
+
+	// Build all the simulations.
+	size_t config_index = 0;
+	std::vector<std::tuple<
+	    std::string, std::string, SingleSimulationConfig,
+	    std::shared_ptr<multiregion::SimulationTask<StrideSimulatorResult>>>>
```

Comment:
This `std::tuple` deserves to be its own struct with descriptive record names, I feel.